### PR TITLE
feat(proxy): [BACK-1392 + BACK-1409] Call Client API and serve REST data

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Request scheduled surface items for a given date and serve it through a REST proxy for Braze.

- Taking a response from Client API and serving all the requested  data except for the author field (as it's not available yet on the public query).

- Relevant packages installed.

## I'd love feedback/perspectives on:
- Do we need to do any input validation for the two variables in the request (date + scheduled surface ID?) Or are we happy for a request to fail if the variables are not provided in the correct form? Yes. Note: to be done in https://getpocket.atlassian.net/browse/BACK-1393

## References

JIRA ticket:
* https://getpocket.atlassian.net/browse/BACK-1392
* https://getpocket.atlassian.net/browse/BACK-1409

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,17 +23,20 @@
...
     "apollo-server-plugin-response-cache": "^3.3.0",
     "aws-xray-sdk-express": "^3.3.4",
-    "dataloader": "^2.0.0",
-    "graphql": "^16.3.0",
-    "graphql-tag": "^2.12.6",
     "cross-fetch": "3.1.5",
-    "express": "4.17.3"
+    "dataloader": "^2.0.0",
+    "express": "4.17.3",
+    "graphql": "^15.8.0",
```
Comment (nina-py) [inline package.json:38]: Had to downgrade the `graphql` package to be able to use `apollo-link-http`.

File: src/main.ts [hunk 2]
```diff
@@ -41,16 +42,13 @@ app.get('/.well-known/server-health', (req, res) => {
   res.status(200).send('ok');
 });
 
-// TODO: return appropriate cache headers here
-// let's start with 30 minutes
-// need to research exact header names
-app.get('/', async (req, res) => {
+app.get('/:scheduledSurfaceID/:date', async (req, res) => {
```
Comment (nina-py) [inline src/main.ts:45]: URL structure here is completely arbitrary, open to suggestions to make it more readable, maybe `/surface/:scheduledSurfaceID/date/:date`?

File: package.json (hunk 1 above)
Comment (bassrock) [reply package.json:38]: Heads up that apollo-link is deprecated https://github.com/apollographql/apollo-link They recommend using apollo client and the packages are moved into there. I don't think you need to update it now in this pr, but something to keep in mind for future work.

**Review by Herraj — DISMISSED** [review]
Looks good to me!

File: src/main.ts [hunk 3]
```diff
@@ -41,16 +42,13 @@ app.get('/.well-known/server-health', (req, res) => {
...
   // enable 30 minute cache when in AWS
   if (config.app.environment !== 'development') {
     res.set('Cache-control', 'public, max-age=1800');
   }
 
-  res.json({ Hello: 'World' });
+  res.json(await getStories(req.params.date, req.params.scheduledSurfaceID));
```
Comment (jpetto) [inline src/main.ts:51]: not a blocker for this PR, but should we do any validation on the `date` parameter? could do a regex right in the url definition...

File: package.json (hunk 1 above)
Comment (jpetto) [reply package.json:38]: looks like this is the new way? https://www.apollographql.com/docs/react/api/link/apollo-link-http/

would be good to update our other repos, too.

**Review by jpetto — APPROVED** [review]
looks good to me! i think we can do url param validation in a follow up.

File: src/main.ts (hunk 2 above)
Comment (jpetto) [reply src/main.ts:45]: i am all for following conventions

File: src/main.ts (hunk 3 above)
Comment (nina-py) [reply src/main.ts:51]: I did ask for feedback on that in the PR! :) I've added date validation to the ticket that should limit the data to Pocket Hits surfaces only: https://getpocket.atlassian.net/browse/BACK-1393

File: package.json (hunk 1 above)
Comment (nina-py) [reply package.json]: Fixed in this PR - as far as I know, we have this setup in the FF Android Proxy repo as well - it should be a quick fix to update it there.

**Pull request merged**